compiler/cpp/src/thrift/generate/t_go_generator.cc

```diff
@@ -2091,8 +2108,11 @@ void t_go_generator::generate_service_client(t_service* tservice) {
       std::string resultName = tmp("_result");
       std::string resultType = publicize(method + "_result", true);
       f_types_ << indent() << "var " << resultName << " " << resultType << endl;
-      f_types_ << indent() << "if err = p.Client_().Call(ctx, \""
-        << method << "\", &" << argsName << ", &" << resultName << "); err != nil {" << endl;
+      f_types_ << indent() << "var meta thrift.ResponseMeta" << endl;
+      f_types_ << indent() << "meta, err = p.Client_().Call(ctx, \""
+        << method << "\", &" << argsName << ", &" << resultName << ")" << endl;
+      f_types_ << indent() << "p.SetLastResponseMeta_(meta)" << endl;
+      f_types_ << indent() << "if err != nil {" << endl;
```

Comment:
should we use `tmp("_meta")` or something? I'm seeing `meta` redeclaration errors when `meta` is already defined in args, for example:
https://github.com/apache/storm/blob/v2.2.0/storm-client/src/storm.thrift#L752
produces the snippet below which throws `meta` redeclared error
```go
// Parameters:
//  - Key
//  - Meta
func (p *NimbusClient) BeginCreateBlob(ctx context.Context, key string, meta *SettableBlobMeta) (r string, err error) {
  var _args293 NimbusBeginCreateBlobArgs
  _args293.Key = key
  _args293.Meta = meta
  var _result294 NimbusBeginCreateBlobResult
  var meta thrift.ResponseMeta
  meta, err = p.Client_().Call(ctx, "beginCreateBlob", &_args293, &_result294)
  p.SetLastResponseMeta_(meta)
  if err != nil {
    return
  }
  switch {
  case _result294.Aze!= nil:
    return r, _result294.Aze
  case _result294.Kae!= nil:
    return r, _result294.Kae
  }

  return _result294.GetSuccess(), nil
}
```

Reply:
Yes. Thanks for reporting. Preparing a PR to fix it.

Reply:
https://github.com/apache/thrift/pull/2332